server/src/services/plugin.service.ts

```diff
@@ -0,0 +1,317 @@
+import { Plugin as ExtismPlugin, newPlugin } from '@extism/extism';
+import { BadRequestException, Injectable } from '@nestjs/common';
+import { validateOrReject } from 'class-validator';
+import { join } from 'node:path';
+import { Asset, WorkflowAction, WorkflowFilter } from 'src/database';
+import { OnEvent, OnJob } from 'src/decorators';
+import { PluginManifestDto } from 'src/dtos/plugin-manifest.dto';
+import { mapPlugin, PluginResponseDto } from 'src/dtos/plugin.dto';
+import { JobName, JobStatus, PluginTriggerType, QueueName } from 'src/enum';
+import { ArgOf } from 'src/repositories/event.repository';
+import { BaseService } from 'src/services/base.service';
+import { PluginHostFunctions } from 'src/services/plugin-host.functions';
+import { IWorkflowJob, JobItem, JobOf, WorkflowData } from 'src/types';
+import { TriggerConfig } from 'src/types/plugin-schema.types';
+
+interface WorkflowContext {
+  authToken: string;
+  asset: Asset;
+  triggerConfig: TriggerConfig | null;
+}
+
+interface PluginInput<T = unknown> {
+  authToken: string;
+  config: T;
+  data: {
+    asset: Asset;
+  };
+}
+
+@Injectable()
+export class PluginService extends BaseService {
+  private pluginJwtSecret!: string;
+  private loadedPlugins: Map<string, ExtismPlugin> = new Map();
+  private hostFunctions!: PluginHostFunctions;
+
+  @OnEvent({ name: 'AppBootstrap' })
+  async onBootstrap() {
+    this.pluginJwtSecret = this.cryptoRepository.randomBytesAsText(32);
+
+    await this.loadPluginsFromManifests();
+
+    this.hostFunctions = new PluginHostFunctions(
+      this.assetRepository,
+      this.albumRepository,
+      this.accessRepository,
+      this.cryptoRepository,
+      this.logger,
+      this.pluginJwtSecret,
+    );
+
+    await this.loadPlugins();
+  }
+
+  //
+  // CRUD operations for plugins
+  //
+  async getAll(): Promise<PluginResponseDto[]> {
+    const plugins = await this.pluginRepository.getAllPlugins();
+    return plugins.map((plugin) => mapPlugin(plugin));
+  }
+
+  async get(id: string): Promise<PluginResponseDto> {
+    const plugin = await this.pluginRepository.getPlugin(id);
+    if (!plugin) {
+      throw new BadRequestException('Plugin not found');
+    }
+    return mapPlugin(plugin);
+  }
+
+  ///////////////////////////////////////////
+  // Plugin Loader
+  //////////////////////////////////////////
+  async loadPluginsFromManifests(): Promise<void> {
+    // Load core plugin
+    const { resourcePaths, plugins } = this.configRepository.getEnv();
+    const coreManifestPath = `${resourcePaths.corePlugin}/manifest.json`;
+
+    const coreManifest = await this.pluginRepository.readManifest(coreManifestPath);
```

Comment:
this should just use the file system/storage repository to read a file and you should JSON.parse and validate it in the service method here.